ocaml/xapi/xapi_pool_update.ml

```diff
@@ -310,7 +309,10 @@ let guidance_from_string = function
 let parse_update_info xml =
   match xml with
   | Xml.Element ("update", attr, children) ->
-      let key = try List.assoc "key" attr with _ -> "" in
+      let key =
+        List.assoc_opt "key" attr |> fun k ->
+        Option.bind k (function "" -> None | s -> Some (Filename.basename s))
```

Comment:
Using `(List.assoc_opt "key" attr)` directly as the first parameter for the bind seems like a simpler way to code this.